src/components/form/ContourForm.tsx

```diff
@@ -0,0 +1,242 @@
+  };
+
+  const removeRoute = (index: number) => {
+    const newRoutes = formData.routes.filter((_, i) => i !== index);
+    updateFormData((prev) => ({ ...prev, routes: newRoutes }));
+  };
+
+  return (
+    <>
+      <Form onSubmit={onSubmit}>
+        {(validationErrors.length > 0 || saveError) && (
+          <Alert
+            variant="danger"
+            title={saveError || t('validation_errors')}
+            isInline
+          >
+            {validationErrors.length > 0 && (
+              <ul>
+                {validationErrors.map((error, index) => (
+                  <li key={index}>{error}</li>
+                ))}
+              </ul>
+            )}
+          </Alert>
+        )}
+
+        <FormGroup
+          className="pf-u-mt-md"
+          fieldId="name"
+          label={t('name')}
+          isRequired
+        >
+          <TextInput
+            isDisabled={isEdit}
+            value={formData?.name}
+            onChange={(value) =>
+              updateFormData((prev) => ({ ...prev, name: value }))
+            }
+          />
+          <div className="help-block">{t('unique_name_for_Proxy')}</div>
+        </FormGroup>
+        <FormGroup fieldId="proxy_type" label={t('http_proxy_type')} isRequired>
+          <CustomDropdown
+            options={ingressClasses?.map((ic) => ({
+              label: ic.metadata.name,
+              value: ic.metadata.name,
+            }))}
+            value={formData?.ingressClassName}
+            onChange={(value) =>
+              updateFormData((prev) => ({
+                ...prev,
+                ingressClassName: value,
+              }))
+            }
+            placeholder={t('private')}
+          />
+          <div className="help-block">{t('http_proxy_type')}</div>
+        </FormGroup>
+        <FormGroup fieldId="hostname" label={t('hostname')} isRequired>
+          <TextInput
+            value={formData?.fqdn}
+            onChange={(value) => {
+              updateFormData((prev) => ({ ...prev, fqdn: value }));
+            }}
+          />
+          <div className="help-block">{t('hostname_http_proxies')}</div>
+        </FormGroup>
+        <div className="pf-c-form__group ">
+          {formData?.routes?.map((route, index) => (
+            <div key={index} className="pf-u-mb-md">
+              <RouteForm
+                route={route}
+                onChange={(updatedRoute) => updateRoute(index, updatedRoute)}
+                onDelete={
+                  formData?.routes.length > 1
+                    ? () => removeRoute(index)
+                    : undefined
+                }
+                availableServices={services?.map((svc) => svc.metadata.name)}
+                availablePorts={services?.flatMap((svc) =>
+                  svc.spec.ports?.map((port) => port.port.toString()),
+                )}
+                availableSecrets={secrets?.map(
+                  (secret) => secret.metadata.name,
+                )}
+              />
```

Comment:
can we move it to a sub-component?
https://github.com/snapp-incubator/contour-console-plugin/pull/2#issuecomment-2591897860